react/features/speaker-stats/components/SpeakerStatsSearch.js

```diff
@@ -0,0 +1,136 @@
+/* @flow */
+
+import { FieldTextStateless as TextField } from '@atlaskit/field-text';
+import React, { PureComponent } from 'react';
+
+import { translate } from '../../base/i18n';
+import { getFieldValue } from '../../base/react';
+import { connect } from '../../base/redux';
+
+/**
+ * The type of the React {@code Component} props of {@link SpeakerStatsSearch}.
+ */
+type Props = {
+
+    /**
+     * The speakerStatsSearch config setting.
+     */
+    _speakerStatsSearch: Boolean,
+
+    /**
+     * The function to initiate the change in the speaker stats table.
+     */
+    onSearch: Function,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function
+};
+
+type State = {
+
+    /**
+     * The current value of the field.
+     */
+    value: string
+
+};
+
+/**
+ * React component for display an individual user's speaker stats.
+ *
+ * @extends Component
+ */
+class SpeakerStatsSearch extends PureComponent<Props, State> {
```

Comment:
This could probably go ahead and be a function component. `useState` for the local state, a selector and `useSelector` for the config flag and also please wrap the `onChange` handler in a `useCallback`.

Reply:
I agree, it is better with functional components, but all the components I found were class components and I did not know if that would be accepted, I will change it, thank you for your remark.

Reply:
I has been changed